youngseo/app.js

```diff
@@ -48,12 +64,41 @@ app.post("/users/signup", async (req, res) => {
             profile_image
         ) VALUES (?, ?, ?, ?);
         `,
-        [ name, email, password, profile_image]
+        [ name, email, hashedPassword, profile_image]
     ); 
 
     return res.status(201).json({ message: "userCreated" });
 });
 
+app.post("/users/login", async (req, res) => {
+    const { email, password } = req.body
+
+    const [userInfo] = await appDataSource.query(
+        `SELECT 
+            users.email,
+            users.password
+        FROM users
+        WHERE users.email = ?
+        `, [email]
+    );
+
+    const checkHash = async (password, hashedPassword) => {
+        return await bcrypt.compare(password, hashedPassword)
+    }
+
+    if (await checkHash(password, userInfo.password)) {
+        const payLoad = { 
+            email : email,
+            password : userInfo.password
+        };
+        const jwtToken = jwt.sign(payLoad,secretKey);
+        return res.status(201).json({accessToken: jwtToken });
+    } else {
+        return res.status(409).json({ message : "Invalid User" });
+    }
```

Comment:
```suggestion
    const match = await bcrypt.compare(password, hashedPassword)

    if (!match) {
        return res.status(409).json({ message : "Invalid User" })
    };
    
    const accessToken = jwt.sign( . . . )
    
    return res.status(200).json({accessToken})
```
다음과 같이 코드를 줄이면서 가독성 좋게 만들 수 있을 것 같아요! 

영서님께서 작성하신 checkHash 부분(= 제가 남겨드린 코드블럭의 match)을 활용하여 if 조건문을 통해 becrypt.compare()가 falsy한 값일 시에만 적절한 에러처리를 하고, 그렇지 않다면 코드를 계속해서 읽어내려오며 최종적으로는 성공적인 response를 보내주는 것이 낫습니다.

다시 한 번 간단히 말하자면 API의 코드 흐름은 동작하며 중간에 성공적인 response를 반환하고 나머지 경우에 에러를 반환하는 것이 아닌, 모든 에러 혹은 예외들을 다 통과하고 최종적으로 가장 하단의 코드에서 성공 response를 해주는 것이 더 효율적입니다!

다만 코드에는 정답이 없다는 것과 지금은 내 머릿속에 있는 로직을 코드로 구현해내기만 해도 충분히 잘하고 있다는 것을 꼭 알고 넘어가주세요 :)

Reply:
종범님 조언에 맞춰 수정하니 훨씬 깔끔하고 간결해 보이네요. 
선 에러처리 후 성공 response 기억하겠습니다 : )